Imagine/Filter/PostProcessor/JpegOptimPostProcessor.php

```diff
@@ -46,129 +40,127 @@ class JpegOptimPostProcessor implements PostProcessorInterface
+     * @param string $temporaryRootPath Directory where temporary file will be written
      */
-    protected $tempDir;
+    public function __construct($executablePath = '/usr/bin/jpegoptim', $strip = true, $quality = null, $progressive = true, $temporaryRootPath = null)
+    {
+        parent::__construct($executablePath, $temporaryRootPath);
 
-    /**
-     * Constructor.
-     *
-     * @param string $jpegoptimBin Path to the jpegoptim binary
-     * @param bool   $stripAll     Strip all markers from output
-     * @param int    $max          Set maximum image quality factor
-     * @param bool   $progressive  Force output to be progressive
-     * @param string $tempDir      Directory where temporary file will be written
-     */
-    public function __construct(
-        $jpegoptimBin = '/usr/bin/jpegoptim',
-        $stripAll = true,
-        $max = null,
-        $progressive = true,
-        $tempDir = ''
-    ) {
-        $this->jpegoptimBin = $jpegoptimBin;
-        $this->stripAll = $stripAll;
-        $this->max = $max;
+        $this->strip = $strip;
+        $this->quality = $quality;
         $this->progressive = $progressive;
-        $this->tempDir = $tempDir ?: sys_get_temp_dir();
     }
 
     /**
-     * @param int $max
+     * @deprecated All post-processor setters have been deprecated in 2.2 for removal in 3.0. You must only use the
+     *             class's constructor to set the property state.
+     *
+     * @param int $maxQuality
      *
      * @return JpegOptimPostProcessor
      */
-    public function setMax($max)
+    public function setMax($maxQuality)
     {
-        $this->max = $max;
+        $this->triggerSetterMethodDeprecation(__METHOD__);
+        $this->quality = $maxQuality;
 
         return $this;
     }
 
     /**
+     * @deprecated All post-processor setters have been deprecated in 2.2 for removal in 3.0. You must only use the
+     *             class's constructor to set the property state.
+     *
      * @param bool $progressive
      *
      * @return JpegOptimPostProcessor
      */
     public function setProgressive($progressive)
     {
+        $this->triggerSetterMethodDeprecation(__METHOD__);
         $this->progressive = $progressive;
 
         return $this;
     }
 
     /**
-     * @param bool $stripAll
+     * @deprecated All post-processor setters have been deprecated in 2.2 for removal in 3.0. You must only use the
+     *             class's constructor to set the property state.
+     *
+     * @param bool $strip
      *
      * @return JpegOptimPostProcessor
      */
-    public function setStripAll($stripAll)
+    public function setStripAll($strip)
     {
-        $this->stripAll = $stripAll;
+        $this->triggerSetterMethodDeprecation(__METHOD__);
+        $this->strip = $strip;
 
         return $this;
     }
 
-    /**
-     * @param BinaryInterface $binary
-     * @param array           $options
-     *
-     * @throws ProcessFailedException
```

Comment:
i would keep the exception annotations. they add information that is not part of the code.

Reply:
oops, I overlooked this